packages/synapse-react-client/src/components/OAuthClientManagement/CreateOAuthClient.tsx

```diff
@@ -106,31 +101,37 @@ export const CreateOAuthModal: React.FunctionComponent<
   useEffect(() => {
     setClientName(client?.client_name ?? '')
     setRedirectUris(
-      client?.redirect_uris.map(str => ({ uri: str })) ?? [{ uri: '' }],
+      client?.redirect_uris?.map((str: string) => ({ uri: str })) ?? [
+        { uri: '' },
+      ],
     )
     setPolicyUri(client?.policy_uri ?? '')
     setClientUri(client?.client_uri ?? '')
     setSectorUri(client?.sector_identifier_uri ?? '')
     setUserinfoSignedResponseAlg(
-      client?.userinfo_signed_response_alg ??
-        defaultUserInfoSignedResponseAlgorithm,
+      (client?.userinfo_signed_response_alg ??
+        defaultUserInfoSignedResponseAlgorithm) as
+        | 'JSON'
+        | OIDCSigningAlgorithm,
     )
     setTosUri(client?.tos_uri ?? '')
   }, [isShowingModal, client])
 
   useDebouncedEffect(
     () => {
-      if (accessToken && oAuthClient.client_id) {
+      if (oAuthClient.client_id) {
```

Comment:
I don't think you can ever see this component without being logged in, so I'm not worried about the `accessToken` checks